### PR TITLE
CLIC.Subcommand: split global and subcommand switch parsing

### PR DESCRIPTION
**Pull request by Fabien-Chouteau**
With this change, the global command and subcommand can use the same switches and they won't clash.

One of the goal here is to realize the full potential of the Switches_As_Args option where parsing of sub-command switches is disabled.

For example this will allow:
$ alr build -f -XTEST=plop # -f is passed to gprbuild and not to alr
$ alr exec bash -c "echo lol" # -c is passed to bash and not to alr

This will break some command line that used to work, for instance:
$ alr with hello --non-interactive

### CODE REVIEW

**Comment by mosteo** [comment]
So now global switches have to come before the subcommand?

File: src/clic-subcommand-instance.ads [hunk 1]
```diff
@@ -42,15 +42,23 @@ package CLIC.Subcommand.Instance is
...
    --  If Alias is already set, it will be silently replaced.
 
-   procedure Execute;
+   procedure Execute
+     (Command_Line : AAA.Strings.Vector := AAA.Strings.Empty_Vector);
    --  Parse the command line and execute a sub-command or display help/usage
    --  depending on command line args.
+   --
+   --  If Command_Line is not empty it will used instead of the actual command
+   --  line arguments from Ada.Command_Line.
```
Comment (mosteo) [inline src/clic-subcommand-instance.ads:51]: will [be] used? Same in line 60

**Review by mosteo — APPROVED** [review]

File: src/clic-subcommand-instance.ads (hunk 1 above)
Comment (Fabien-Chouteau) [reply src/clic-subcommand-instance.ads:51]: Indeed :+1:

**Comment by Fabien-Chouteau** [comment]
> So now global switches have to come before the subcommand?

Yes, that's a breaking change.
I check and that's how `git` seems to work as well.

**Comment by mosteo** [comment]
> > So now global switches have to come before the subcommand?
> 
> Yes, that's a breaking change.
> I check and that's how `git` seems to work as well.

OK, not a big deal.

**Pull request merged**